sound/soc/sof/intel/hda-stream.c

```diff
@@ -152,20 +152,39 @@ int hda_dsp_stream_spib_config(struct snd_sof_dev *sdev,
 
 /* get next unused stream */
 struct hdac_ext_stream *
-hda_dsp_stream_get(struct snd_sof_dev *sdev, int direction)
+hda_dsp_stream_get(struct snd_sof_dev *sdev, int direction, int flag)
 {
 	struct hdac_bus *bus = sof_to_bus(sdev);
 	struct hdac_ext_stream *stream = NULL;
 	struct hdac_stream *s;
 
 	spin_lock_irq(&bus->reg_lock);
 
-	/* get an unused stream */
-	list_for_each_entry(s, &bus->stream_list, list) {
-		if (s->direction == direction && !s->opened) {
-			s->opened = true;
-			stream = stream_to_hdac_ext_stream(s);
-			break;
+	/*
+	 * Reserved host dma only stream needs to be allocated from tail of
+	 * list which is organized with stream tag value increased from one to
+	 * max, so that normal capture stream could get the same stream tag
+	 * from host and link dma allocator, otherwise, when applying coupled
```

Comment:
Sorry, I don't get why a stream would need to have the same tag on both host and link DMA. This only makes sense if you have a passthrough flow, but the moment you start mixing two streams this assumption is no longer valid. Likewise if we have a stream that originates in the firmware, like tone generator, there will be no host DMA.
What am I missing?